apps/etterlatte-behandling/src/main/kotlin/behandling/tilbakekreving/TilbakekrevingDao.kt

```diff
@@ -25,35 +42,125 @@ class TilbakekrevingDao(private val connection: () -> Connection) {
                     """.trimIndent(),
                 )
             statement.setObject(1, tilbakekrevingId)
-            return statement.executeQuery().singleOrNull {
-                toTilbakekreving()
-            } ?: throw TilbakekrevingFinnesIkkeException("Tilbakekreving med id=$tilbakekrevingId finnes ikke")
+            return statement.executeQuery().singleOrNull { toTilbakekreving(perioder) }
         }
-    }
 
-    fun lagreTilbakekreving(tilbakekreving: Tilbakekreving): Tilbakekreving {
-        with(connection()) {
+    private fun selectTilbakekrevingsperioder(
+        connection: Connection,
+        tilbakekrevingId: UUID,
+    ): List<TilbakekrevingPeriode> =
+        with(connection) {
             val statement =
                 prepareStatement(
                     """
-                    INSERT INTO tilbakekreving(id, status, sak_id, opprettet, kravgrunnlag)
-                    VALUES (?, ?, ?, ?, ?)
-                    ON CONFLICT (id) DO UPDATE SET status = excluded.status, kravgrunnlag = excluded.kravgrunnlag
+                    SELECT * FROM tilbakekrevingsperiode WHERE tilbakekreving_id = ?
                     """.trimIndent(),
                 )
-            statement.setObject(1, tilbakekreving.id)
-            statement.setString(2, tilbakekreving.status.name)
-            statement.setLong(3, tilbakekreving.sak.id)
-            statement.setTidspunkt(4, tilbakekreving.opprettet)
-            statement.setJsonb(5, tilbakekreving.kravgrunnlag.toJsonNode())
-            statement.executeUpdate()
-        }.also {
-            require(it == 1)
-            return hentTilbakekreving(tilbakekreving.id)
+            statement.setObject(1, tilbakekrevingId)
+            val alle = statement.executeQuery().toList { toTilbakekrevingsperiode() }
+            val ytelsebeloeper = alle.filter { it.second.klasseType == KlasseType.YTEL.name }
+            val feilkonto = alle.filter { it.second.klasseType == KlasseType.FEIL.name }
+            return ytelsebeloeper.map { (maaned, ytelsebeloeper) ->
+                TilbakekrevingPeriode(
+                    maaned = maaned,
+                    ytelsebeloeper = ytelsebeloeper,
+                    feilkonto =
+                        feilkonto.find { it.first == maaned }?.second
+                            ?: throw TilbakekrevingHarMangelException("Mangler feilkonto for tilbakekrevingsperiode"),
+                )
+            }
+        }
+
+    fun lagreTilbakekreving(tilbakekreving: Tilbakekreving): Tilbakekreving {
+        with(connection()) {
+            insertTilbakekreving(this, tilbakekreving)
+            tilbakekreving.perioder.forEach {
+                insertTilbakekrevingsperiode(this, tilbakekreving.id, it.maaned, it.ytelsebeloeper)
```

Comment:
Her kan vi kjøre `addBatch` etter hver runde i loopen og avslutte med `executeBatch` for begge disse vel.